ParallelMatrixMultiplication/ParallelMatrixMultiplication/Matrix.cs

```diff
@@ -0,0 +1,87 @@
+namespace ParallelMatrixMultiplication;
+
+public class Matrix
+{
+    private readonly int[,] _matrixArray;
+    private readonly int _numOfRows;
+    private readonly int _numOfCols;
+
+    //constructor
+    public Matrix(string filepath)
+    {
+        _matrixArray = convertFileToMatrix(filepath);
+        _numOfRows = _matrixArray.GetLength(0);
+        _numOfCols = _matrixArray.GetLength(1);
+        
+    }
+    
+    public Matrix(int numOfRows, int numOfCols)
+    {
+        _matrixArray = new int[numOfRows, numOfCols];
+        _numOfRows = numOfRows;
+        _numOfCols = numOfCols;
+        
+    }
+    
+    //indexer
+    public int this[int _numOfRows, int _numOfCols ]
+    {
+        get { return _matrixArray[_numOfRows, _numOfCols]; }
+        set { _matrixArray[_numOfRows, _numOfCols] = value; }
+    }
+    
+    //properties
+    public int NumOfRows => _numOfRows;
+    public int NumOfCols => _numOfCols;
+    
+    private int[,] convertFileToMatrix(string filepath)
+    {
+       
+        string[] lines = File.ReadAllLines(filepath);
+
+        if (lines.Length == 0)
+        {
+            throw new ArgumentException("The file path is empty.");
+        }
+        
+       
+        int numOfRows = lines.Length;
+        int numOfCols = lines[0].Split(' ').Length;
+
+        
+        int[,] matrix = new int[numOfRows, numOfCols];
```

Comment:
```suggestion
        var matrix = new int[numOfRows, numOfCols];
```
Не пишите в объявлении один тип дважды